src/helpers/commandHelper.ts

```diff
@@ -89,9 +89,14 @@ export class CommandEnvironmentHelper {
     if (vscode.workspace.workspaceFolders !== undefined) {
       args.push('--local-workspace=' + vscode.workspace.workspaceFolders[0].uri.fsPath);
     }
-    if (settings.editorService.modulePath !== undefined && settings.editorService.modulePath !== '') {
-      args.push('--puppet-settings=--modulepath,' + settings.editorService.modulePath);
+
+    let puppetSettings: string[] = [];
+    if (settings.editorService.additionalPuppetSettings !== undefined && settings.editorService.additionalPuppetSettings !== '') { puppetSettings.push(settings.editorService.additionalPuppetSettings); }
+    if (settings.editorService.modulePath !== undefined && settings.editorService.modulePath !== '') { puppetSettings.push('--modulepath,' + settings.editorService.modulePath); }
```

Comment:
Since we have modulepath already, is it too much/ or too fragile to ask the user to make a properly formatted csv line? Should we instead add options we know work as configuration points like modulepath? Or go for an array like for `featureFlags`?

Reply:
Interesting.  I went this route as I didn't know what advanced settings users would need.  So I defaulted to the raw setting instead of nice abstracted view.

The only ones I know of are `--confdir` and `--vardir`.  Possibly `--environment` too.

Reply:
I hate to expand scope on you, but I think since we're here editing this anyway, it would make sense to make this a little more discoverable and less bag o settings type thing. I vote to have this be the ones you listed above plus `modulepath` as distinct settings.

Reply:
Do you think it should be

`puppet.editorService.confdir`
`puppet.editorService.puppet_confdir`

or perhaps it's own collection (May require modulepath to be moved)
`puppet.editorService.puppet.confdir`

Reply:
OKay...I'll need setup a deprecation of modulepath too.  Not a big deal.